Client/Application/AppDelegate.swift

```diff
@@ -218,43 +218,50 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject) -> Bool {
-        if let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) {
-            if components.scheme != "firefox" && components.scheme != "firefox-x-callback" {
-                return false
-            }
-            var url: String?
-            var isPrivate: Bool = false
-            for item in (components.queryItems ?? []) as [NSURLQueryItem] {
-                switch item.name {
-                case "url":
-                    url = item.value
-                case "private":
-                    isPrivate = (item.value != "false" && item.value != "0")
-                default: ()
-                }
+        guard let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) else {
```

Comment:
we combine this into one statement

`guard let components = NSURLComponents(URL: url, resolvingAgainstBaseURL: false) where components.scheme != "firefox" && components.scheme != "firefox-x-callback" else { return false } `